src/components/calcite-time-picker/calcite-time-picker.tsx

```diff
@@ -0,0 +1,702 @@
+            } else {
+              this.minuteEl.focus();
+            }
+            break;
+        }
+        break;
+    }
+  }
+
+  //--------------------------------------------------------------------------
+  //
+  //  Public Methods
+  //
+  //--------------------------------------------------------------------------
+
+  @Method()
+  async setFocus(): Promise<void> {
+    this.hourEl.focus();
+  }
+
+  // --------------------------------------------------------------------------
+  //
+  //  Private Methods
+  //
+  // --------------------------------------------------------------------------
+
+  private amPmKeyDownHandler = (event: KeyboardEvent): void => {
+    switch (event.key) {
+      case "a":
+        this.setTime("ampm", "AM");
+        break;
+      case "p":
+        this.setTime("ampm", "PM");
+        break;
+      case "Backspace":
+        this.setTime("ampm", "--");
+        break;
+      case "ArrowUp":
+        event.preventDefault();
+        this.incrementAmPm();
+        break;
+      case "ArrowDown":
+        event.preventDefault();
+        this.decrementAmPm();
+        break;
+    }
+  };
+
+  private decrementAmPm = (event?: KeyboardEvent | MouseEvent): void => {
+    if (event && event instanceof KeyboardEvent && event.key !== "Enter") return;
+    switch (this.ampm) {
+      case "--":
+      case "AM":
+        this.setTime("ampm", "PM");
+        break;
+      case "PM":
+        this.setTime("ampm", "AM");
+        break;
+    }
+  };
+
+  private decrementHour = (event?: KeyboardEvent | MouseEvent): void => {
+    if (event && event instanceof KeyboardEvent && event.key !== "Enter") return;
+    switch (this.hour) {
+      case "--":
+        this.setTime("hour", 0);
+        break;
+      case "00":
+        this.setTime("hour", 23);
+        break;
+      default:
+        this.setTime("hour", parseInt(this.hour) - 1);
+        break;
+    }
+  };
+
+  private decrementMinuteOrSecond = (key: MinuteOrSecond): void => {
+    if (this[key] === "--") {
+      this.setTime(key, 59);
+    } else {
+      const valueAsNumber = parseInt(this[key]);
+      if (valueAsNumber === 0) {
+        this.setTime(key, 59);
+      } else {
+        this.setTime(key, valueAsNumber - 1);
+      }
+    }
+  };
+
+  private decrementMinute = (event?: KeyboardEvent | MouseEvent): void => {
+    if (event && event instanceof KeyboardEvent && event.key !== "Enter") return;
+    this.decrementMinuteOrSecond("minute");
+  };
+
+  private decrementSecond = (event?: KeyboardEvent | MouseEvent): void => {
+    if (event && event instanceof KeyboardEvent && event.key !== "Enter") return;
+    this.decrementMinuteOrSecond("second");
+  };
+
+  private focusHandler = (event: FocusEvent): void => {
+    this.activeEl = event.target as HTMLSpanElement;
+  };
+
+  private getAmPm = (): AmPm => {
+    if (this.hour === "--") return "--";
+    const hourAsNumber = parseInt(this.hour);
+    if (hourAsNumber >= 0 && hourAsNumber <= 11) {
```

Comment:
Simplify to `return hourAsNumber >= 0 && hourAsNumber <= 11 ? "AM" : "PM";`

Applies to other sections.